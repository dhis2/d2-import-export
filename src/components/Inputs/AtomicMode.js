import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { atomicModeOptions } from '../../utils/options'
import { RadioGroupField } from '../RadioGroup'

const NAME = 'atomicMode'
const DATATEST = 'input-atomic-mode'
const LABEL = i18n.t('Atomic mode')

const AtomicMode = () => (
    <RadioGroupField
        name={NAME}
        label={LABEL}
        options={atomicModeOptions}
        dataTest={DATATEST}
    />
)

export { AtomicMode }
