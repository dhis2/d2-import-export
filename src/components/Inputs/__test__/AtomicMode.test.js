import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { AtomicMode } from '../AtomicMode'

jest.mock('react-final-form')

describe('Input component - AtomicMode', () => {
    it('should render correctly', () => {
        const file = shallow(<AtomicMode />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
