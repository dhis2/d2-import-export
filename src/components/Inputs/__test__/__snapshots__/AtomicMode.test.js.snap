// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - AtomicMode should render correctly 1`] = `
<Field>
  <Label>
    Atomic mode
  </Label>
  <RadioGroup
    dataTest="input-atomic-mode"
    label="Atomic mode"
    name="atomicMode"
    options={
      Array [
        Object {
          "label": "All",
          "value": "ALL",
        },
        Object {
          "label": "None",
          "value": "NONE",
        },
      ]
    }
  />
</Field>
`;
